client/blocks/importer/wordpress/import-everything/pre-migration/index.tsx

```diff
@@ -231,6 +235,13 @@ export const PreMigrationScreen: React.FunctionComponent< PreMigrationProps > =
 			return <LoadingEllipsis />;
 		}
 
+		if ( ! sourceSite || ( sourceSite && sourceSite.ID !== sourceSiteId ) ) {
```

Comment:
We probably should split these screens into more granular ones, this much of conditional logic is very hard to reason about. But that is not a criticism of this PR, just the design that has evolved.

Reply:
> We probably should split these screens into more granular ones, this much of conditional logic is very hard to reason about. But that is not a criticism of this PR, just the design that has evolved.

Yeah, I agree. We probably want to split those components into its own and tidy it up in the future.